beaker/client/api_providers_test.py

```diff
@@ -0,0 +1,49 @@
+from beaker.client.api_providers import (
+    Network,
+    Sandbox,
+    AlgoNode,
+    AlgoExplorer,
+    PureStake,
+)
+
+
+def test_sandbox():
+    sb = Sandbox(Network.SandNet)
+    sb.algod().suggested_params()
+    # sandbox indexer off
+    # sb.indexer().health()
+
+
+def test_algonode():
+    for network in Network:
+        if network == Network.SandNet:
+            continue
+        print(f"trying {network}")
+        an = AlgoNode(network)
+        an.algod().suggested_params()
+        an.indexer().health()
+
+
+def test_algoexplorer():
+    for network in Network:
+        if network == Network.SandNet:
+            continue
+
+        print(f"trying {network}")
+        ae = AlgoExplorer(network)
+        ae.algod().suggested_params()
+        ae.indexer().health()
+
+
+def test_purestake():
+    return
+
+    api_key = "TODO"
+    for network in Network:
+        if network == Network.SandNet:
+            continue
+
+        print(f"trying {network}")
+        ae = PureStake(network)
+        ae.algod(api_key).suggested_params()
```

Comment:
It would be cool if the API key was given upon initialization of the API provider

Reply:
why prefer that over the current way? could there be diff keys per client type?

Reply:
Ah yes true didn't think about different API keys